src/libcharon/plugins/kernel_super/system_events/sl_system_events/sl_system_events.c

```diff
@@ -0,0 +1,153 @@
+{
+	return g_system_events->ops.libsystem_events_send_ipsec_tunnel_establishment_failed(
+		error_reason);
+}
+static int sl_send_ipsec_tunnel_activated(const char *device_id)
+{
+	return g_system_events->ops.libsystem_events_send_ipsec_tunnel_activated(device_id);
+}
+static int sl_send_ipsec_tunnel_tunnel_deactivated(const char *device_id)
+{
+	return g_system_events->ops.libsystem_events_send_ipsec_tunnel_deactivated(device_id);
+}
+static int sl_send_ipsec_tunnel_down(const char *device_id, const char *error_reason)
+{
+	return g_system_events->ops.libsystem_events_send_ipsec_tunnel_down(device_id, error_reason);
+}
+static int sl_send_ipsec_terminator_up(const char *terminator_id)
+{
+	return g_system_events->ops.libsystem_events_send_ipsec_terminator_up(terminator_id);
+}
+static int sl_send_ipsec_terminator_down(const char *terminator_id, const char *error_reason)
+{
+	return g_system_events->ops.libsystem_events_send_ipsec_terminator_down(terminator_id,
+																			error_reason);
+}
+
+static void sl_system_events_destroy(system_events_obj_t *_this)
+{
+	(void)_this;
+	dlclose(g_system_events->dl_handle);
+	free(g_system_events);
+}
+
+static bool _set_libsystem_events_ops(private_sl_system_events_t *this)
+{
+#define LOAD_LIBSYSTEMEVENTS_SYMBOL(symbol_name)                      \
+	do                                                                \
+	{                                                                 \
+		this->ops.symbol_name = dlsym(this->dl_handle, #symbol_name); \
+		if (! this->ops.symbol_name)                                  \
+		{                                                             \
+			LOG(DBG_KNL, SL_LOG_ERR,                                  \
+				"dlsym() failed to load symbol " #symbol_name         \
+				" from libsystem_events (%s). error=%s",              \
+				LIBSYSTEMEVENTS_SO_PATH, dlerror());                  \
+			return false;                                             \
+		}                                                             \
+	} while (false)
+
+	LOAD_LIBSYSTEMEVENTS_SYMBOL(libsystem_events_send_ipsec_tunnel_established);
+	LOAD_LIBSYSTEMEVENTS_SYMBOL(libsystem_events_send_ipsec_tunnel_establishment_failed);
+	LOAD_LIBSYSTEMEVENTS_SYMBOL(libsystem_events_send_ipsec_tunnel_activated);
+	LOAD_LIBSYSTEMEVENTS_SYMBOL(libsystem_events_send_ipsec_tunnel_deactivated);
+	LOAD_LIBSYSTEMEVENTS_SYMBOL(libsystem_events_send_ipsec_tunnel_down);
+	LOAD_LIBSYSTEMEVENTS_SYMBOL(libsystem_events_send_ipsec_terminator_up);
+	LOAD_LIBSYSTEMEVENTS_SYMBOL(libsystem_events_send_ipsec_terminator_down);
+
+#undef LOAD_LIBSYSTEMEVENTS_SYMBOL
+	return true;
+}
+
+sl_system_events_obj_t *sl_system_events_obj_create(void)
+{
+	LOG(DBG_KNL, SL_LOG_INFO, "initializing sl_system_events");
+
+	if (g_system_events)
+	{
+		LOG(DBG_KNL, SL_LOG_ERR, "sl_system_events already created");
+		return false;
+	}
+	private_sl_system_events_t *this = NULL;
+	INIT(this,
+		 .public = { .interface = {
```

Comment:
Try to manually reformat this block (clang-format it sometimes annoying and manually restructuring the code helps).
If it doesn't work then surround this block with `clang-format off/on`